File: api/src/Workshop.Api/Data/Entities/ReservableObject.cs
namespace Workshop.Api.Data.Entities;

public class ReservableObject
{
    public int Id { get; set; }
    public int LocationId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Type { get; set; } = string.Empty;
    public bool IsAvailable { get; set; } = true;
    public bool HasTimeRestrictions { get; set; }
    public TimeOnly? AvailableFrom { get; set; }
    public TimeOnly? AvailableUntil { get; set; }
    public string? DaysOfWeek { get; set; }
    
    // Navigation property
    public Location Location { get; set; } = null!;
}

File: client/src/app/models/reservable-object.ts
export interface ReservableObject {
  id: number;
  name: string;
  type: string;
  isAvailable: boolean;
  hasTimeRestrictions: boolean;
  availableFrom?: string;
  availableUntil?: string;
  daysOfWeek?: string;
  positionX?: number;
  positionY?: number;
}

export interface ReservableObjectListResponse {
  totalCount: number;
  objects: ReservableObject[];
}

export const ObjectTypes = {
  Desk: 'Desk',
  ParkingSpace: 'ParkingSpace',
  All: 'All'
} as const;

export type ObjectType = typeof ObjectTypes[keyof typeof ObjectTypes];
